test-support/page-object/custom-helper.js

```diff
@@ -0,0 +1,30 @@
+import Ember from 'ember';
+import { build } from './build';
+import { qualifySelector } from './helpers';
+
+export function customHelper(userDefinedFunction) {
+  return function(selector, options) {
+    // def
+    return {
+      buildPageObjectAttribute: function(key, parent) {
+        // def
+        return function() {
+          // eval
+          let scopedSelector = qualifySelector(parent.scope, selector);
+
+          let customHelperRes = userDefinedFunction(scopedSelector, options);
+
+          if ($.isPlainObject(customHelperRes)) {
+            return build($.extend(customHelperRes, { scope: scopedSelector }),
+                         key,
+                         parent);
```

Comment:
I think the order of the parameters in the `extend` is not the best.

If the customHelperRes has a `scope: 'something'` attribute, we're overriding it with our `{ scope: scopedSelector }` and may not be the desired effect.

I think the order could be `$.extend({ scope: scopedSelector }, customHelperRes)`, this way the user can override the scope.